includes/templates/class-amp-post-template.php

```diff
@@ -1,96 +1,52 @@
-
-class AMP_Post_Template {
-	const SITE_ICON_SIZE = 32;
-	const CONTENT_MAX_WIDTH = 600;
-
-	// Needed for 0.3 back-compat
-	const DEFAULT_NAVBAR_BACKGROUND = '#0a89c0';
-	const DEFAULT_NAVBAR_COLOR = '#fff';
-
-	private $template_dir;
-	private $data;
+require_once ( AMP__DIR__ . '/includes/class-amp-content.php' );
 
-	public function __construct( $post_id ) {
-		$this->template_dir = apply_filters( 'amp_post_template_dir', AMP__DIR__ . '/templates' );
-
-		$this->ID = $post_id;
-		$this->post = get_post( $post_id );
-
-		$content_max_width = self::CONTENT_MAX_WIDTH;
-		if ( isset( $GLOBALS['content_width'] ) && $GLOBALS['content_width'] > 0 ) {
-			$content_max_width = $GLOBALS['content_width'];
-		}
-		$content_max_width = apply_filters( 'amp_content_max_width', $content_max_width );
-
-		$this->data = array(
-			'content_max_width' => $content_max_width,
-
-			'document_title' => function_exists( 'wp_get_document_title' ) ? wp_get_document_title() : wp_title( '', false ), // back-compat with 4.3
-			'canonical_url' => get_permalink( $post_id ),
-			'home_url' => home_url(),
-			'blog_name' => get_bloginfo( 'name' ),
-
-			'html_tag_attributes' => array(),
-			'body_class' => '',
+// Require AMP filters
+require_once( AMP__DIR__ . '/includes/sanitizers/require-sanitizers.php' );
+// Require embed classes
+require_once ( AMP__DIR__ . '/includes/embeds/require-embeds.php' );
+// Require base AMP template class
+require_once ( AMP__DIR__ . '/includes/templates/class-amp-template.php' );
 
-			'site_icon_url' => apply_filters( 'amp_site_icon_url', function_exists( 'get_site_icon_url' ) ? get_site_icon_url( self::SITE_ICON_SIZE ) : '' ),
-			'placeholder_image_url' => amp_get_asset_url( 'images/placeholder-icon.png' ),
+class AMP_Post_Template extends AMP_Template {
 
-			'featured_image' => false,
-			'comments_link_url' => false,
-			'comments_link_text' => false,
-
-			'amp_runtime_script' => 'https://cdn.ampproject.org/v0.js',
-			'amp_component_scripts' => array(),
-
-			'customizer_settings' => array(),
-
-			'font_urls' => array(
-				'merriweather' => 'https://fonts.googleapis.com/css?family=Merriweather:400,400italic,700,700italic',
-			),
+	public function __construct( $post_id ) {
 
-			'post_amp_styles' => array(),
-
-			/**
-			 * Add amp-analytics tags.
-			 *
-			 * This filter allows you to easily insert any amp-analytics tags without needing much heavy lifting.
-			 *
-			 * @since 0.4
-			 *.
-			 * @param	array	$analytics	An associative array of the analytics entries we want to output. Each array entry must have a unique key, and the value should be an array with the following keys: `type`, `attributes`, `script_data`. See readme for more details.
-			 * @param	object	$post	The current post.
-			 */
-			'amp_analytics' => apply_filters( 'amp_post_template_analytics', array(), $this->post ),
-			);
+		parent::__construct();
+		
+		$this->post_id = $post_id;
```

Comment:
We should keep $this->ID for back-compat (or maybe add a magic `__get` method to handle this?)